Clever/clever-cli
# dump all the data from the Clever API to a CSV file

There are two parts to this:
1. A subpkg `clevertable` that wraps `clever-go` and satisfies the `optimus.Table` interface, allowing easy concurrent streaming from our API
2. A command line tool that will dump data from our API via a configurable set of options and actions

``` shell
$ clever-cli --token=DEMO_TOKEN teachers list
```

``` shell
$ clever-cli --token=DEMO_TOKEN teachers list '{"name.first":"John"}'
```

Todo for this PR: tests, readme

Future PRs should:
1. add additional output formats (currently only supports csv)
2. add `get` action to get a single resource, e.g. `clever-cli teachers get 4fffff....`
3. always output to stdout. Then csv would look like `clever-cli --output=csv teachers list > teachers.csv`
